### PR TITLE
Clear the global out of date configs if they were previously set

### PR DESCRIPTION
**Pull request by oconnor663**
Fixes https://github.com/keybase/client/issues/9579.

r? @maxtaco

### CODE REVIEW

File: go/libkb/api.go [hunk 1]
```diff
@@ -547,6 +547,17 @@ func (a *InternalAPIEngine) consumeHeaders(resp *http.Response) (err error) {
 			*g.lastUpgradeWarning = now
 		}
 		g.oodiMu.Unlock()
+	} else {
+		// We might be in a state where the server *used to* think we were out
+		// of date, but now it doesn't. (Maybe a bad config got pushed and then
+		// later fixed.) If so, we need to clear the global outOfDateInfo, so
+		// that the client stops printing warnings.
+		g := a.G()
+		g.oodiMu.Lock()
```
Comment (maxtaco) [inline go/libkb/api.go]: can you make a `Clear` method?

**Comment by oconnor663** [comment]
I've been staring at this for a little while, and I'm not sure I can improve on what's there. Everything's protected by this shared lock, and encapsulating that is hard to do without creating another one of those `Account`-style closure-taking APIs that we hate. In this case, even moving all of these updates into a single function with a consolidated `Lock`/`Unlock` is hard, because this `defer a.G().NotifyRouter.HandleClientOutOfDate` needs to happen _after_ unlock (presumably because it re-acquires the same lock via `G.GetOutOfDateInfo` somewhere), and that means the unlock can't be deferred from the most natural place. We could put that call on another goroutine, but then I feel like I'm complicating things instead of simplifying them :|

Even something as simple as putting the three member clears behind a `Clear` function seems unfortunately complicated. Either that function takes the lock itself (and therefore lives on `G` rather than on `GetOutOfDateInfo` where it should), or it doesn't take the lock and then it's a footgun that looks like it's safe to call in places it isn't.

I'm going to run with what's here unless there's a different approach you think I should try?

**Comment by maxtaco** [comment]
I just meant put those 3 `= ""` lines into a subfunction is all

**Comment by maxtaco** [comment]
leave it locked as before...

**Comment by maxtaco** [comment]
but i really don't care either way

**Pull request closed without merging**